packages/block-library/src/navigation-menu/index.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Server-side rendering of the `core/navigation-menu` block.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Renders the `core/navigation-menu` block on server.
+ *
+ * @param array $attributes The block attributes.
+ * @param array $content The saved content.
+ * @param array $block The parsed block.
+ *
+ * @return string Returns the post content with the legacy widget added.
+ */
+function render_block_navigation_menu( $attributes, $content, $block ) {
+	return prepare_navigation( $block );
+}
+
+/**
+ * Walks the inner block structure and retunrs an HTML list of it.
+ *
+ * @param array $block The block.
+ *
+ * @return string Returns  an HTML list from innerBlocks.
+ */
+function prepare_navigation( $block ) {
+	$block_tree = '';
+	foreach ( (array) $block['innerBlocks'] as $key => $menu_item ) {
+		$block_tree .= '<li>' . $menu_item['innerContent'][0];
+		if ( count( (array) $menu_item['innerBlocks'] ) > 0 ) {
```

Comment:
Do we need to cast `$menu_item['innerBlocks']` to an array here?

Reply:
Yes, if we don't cast it we need to check it is an array or we get notices from foreach.